### PR TITLE
Document ItemList control.

### PR DESCRIPTION
**Pull request by zakstam**
I'v been cautious so far about changing anything code related but i think that i'm going to commit any code changes also from now on (anything non-documentation related) and you can stop me at anytime.

### CODE REVIEW

File: scene/gui/item_list.cpp [hunk 1]
```diff
@@ -395,8 +395,10 @@ bool ItemList::is_same_column_width() const {
 }
 
 void ItemList::set_max_text_lines(int p_lines) {
-
```
Comment (groud) [inline scene/gui/item_list.cpp]: This is definitely not how to solve the error. If error are triggered that's normal, it means the calling code should check the p_lines value by itself. The the ERR_FAIL_COND is only a barrier avoiding crashes.

File: scene/gui/item_list.cpp [hunk 2]
```diff
@@ -578,8 +578,10 @@ void ItemList::_gui_input(const Ref<InputEvent> &p_event) {
 			return;
 		}
 
-		// Since closest is null, more likely we clicked on empty space, so send signal to interested controls. Allows, for example, implement items deselecting.
-		emit_signal("nothing_selected");
+		//Since closest is null, more likely we clicked on empty space, so send signal to interested controls. Allows, for example, implement items deselecting.
+		//Pretty much the same as "rmb_clicked" for left mouse click. This could be changed to "mouse_click" with an argument that represents the mouse button index.
+		//example: emit_signal("mouse_click", mb->get_position(), mb->get_button_index())
+		emit_signal("nothing_selected");
```
Comment (bruvzg) [inline scene/gui/item_list.cpp]: Remove trailing whitespace, see [Code style guidelines](https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html).

File: scene/gui/item_list.cpp [hunk 3]
```diff
@@ -372,7 +371,7 @@ void ItemList::clear() {
 }
 
 void ItemList::set_fixed_column_width(int p_size) {
-
+	ERR_EXPLAIN("fixed_column_width can not be less than 0.")
```
Comment (YeldhamDev) [inline scene/gui/item_list.cpp]: Please capitalize the first letter in those `ERR_EXPLAIN` messages (here, 398, and 410).

Comment (groud) [reply scene/gui/item_list.cpp]: The variable name should likely not be used here then. Somehow like "The fixed column width cannot be less than 0."

Comment (YeldhamDev) [reply scene/gui/item_list.cpp]: I actually completely missed that this was a variable name. But I agree that the variable should be ditched from the message.

**Comment by zaksnet** [comment]
Sorry, i'm not sure what on earth i have done and deleted all my commits. I will recreate the pull request.

**Pull request closed without merging**